File: 비타알고/hani0903/1등과 2등.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : '../../input.txt';
let input = fs.readFileSync(filePath, 'utf-8').trim();

let idx21 = input.indexOf('21');

if (idx21 !== -1){
    input = input.slice(0, idx21) + input.slice(idx21 + 2);
    let idx12 = input.indexOf('12');

    if(idx12 !== -1){
        console.log('Yes');
    }else{
        console.log('No');
    }
}else{
    console.log('No');
}

File: 비타알고/1등과 2등.py
input_string = input().strip()

found_21 = []
found_12 = []

i = 0
while i < len(input_string) - 1:
  if input_string[i:i + 2] == '21':
    found_21.append(i)
    i += 1
  elif input_string[i:i + 2] == '12':
    found_12.append(i)
    i += 1
  else:
    i += 1

for idx21 in found_21:
  for idx12 in found_12:
    if (idx21 + 2 <= idx12) or (idx12 + 2 <= idx21):
      print('Yes')
      exit()

print('No')
